src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -153,15 +153,17 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 // Dictionary does not guarantee order will be preserved.
                 var groupedDisplayNamesAndValues = new List<KeyValuePair<EnumGroupAndName, string>>();
                 var namesAndValues = new Dictionary<string, string>();
+                var enumLocalizer = _stringLocalizerFactory?.Create(underlyingType);
                 foreach (var name in Enum.GetNames(underlyingType))
                 {
                     var field = underlyingType.GetField(name);
-                    var displayName = GetDisplayName(field);
                     var groupName = GetDisplayGroup(field);
                     var value = ((Enum)field.GetValue(obj: null)).ToString("d");
 
                     groupedDisplayNamesAndValues.Add(new KeyValuePair<EnumGroupAndName, string>(
-                        new EnumGroupAndName(groupName, displayName),
+                        new EnumGroupAndName(
+                            groupName,
+                            () => { return GetDisplayName(field, enumLocalizer); }),
```

Comment:
Minor but ick: Don't be so verbose when `() => GetDisplayName(field, enumLocalizer)` will do.